File: adolfont/exercism/elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RNATranscription.to_rna('ACTG')
  'UGAC'
  """
  @rna_map %{?G => ?C, ?C => ?G, ?T => ?A, ?A => ?U}

  @spec to_rna([char]) :: [char]

  def to_rna(list) do
    Enum.map(list, fn x -> @rna_map[x] end)
  end
end

File: adolfont/exercism/erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

to_rna([]) ->
    [];
to_rna(Strand) ->
    RnaMap = #{$A => $U, $C => $G, $G => $C, $T => $A},
    lists:map(fun (X) ->
                      maps:get(X, RnaMap)
              end,
              Strand).
